provider/openstack/client.go

```diff
@@ -150,11 +150,7 @@ func (c *ClientFactory) getClientState(options ...client.Option) (client.Authent
 
 // getClientByAuthMode creates a new client for the given AuthMode.
 func (c *ClientFactory) getClientByAuthMode(authMode identity.AuthMode, cred identity.Credentials, options ...client.Option) (client.AuthenticatingClient, error) {
-	gooseLogger := gooselogging.LoggoLogger{
-		Logger: loggo.GetLogger("goose"),
-	}
-
-	newClient, err := c.clientFunc(cred, authMode, gooseLogger, c.sslHostnameConfig.SSLHostnameVerification(), c.spec.CACertificates, options...)
+	newClient, err := c.clientFunc(cred, authMode, c.sslHostnameConfig.SSLHostnameVerification(), c.spec.CACertificates, options...)
```

Comment:
Would it be possible to convert the entire arg list into options?

Reply:
Yeah maybe.